packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/location.dart

```diff
@@ -138,3 +142,51 @@ class LatLngBounds {
   @override
   int get hashCode => Object.hash(southwest, northeast);
 }
+
+/// A data point entry for a heatmap.
+/// This is a geographical data point with a weight attribute.
+class WeightedLatLng extends LatLng {
```

Comment:
The class comment sounds like it describes composition rather than inheritance; i.e., that this is a thing comprised of a LatLng and additional information. I would expect this to have a `LatLng` field, not be a `LatLng`. Is there a reason you need inheritance?

Reply:
I was trying to make it easy for people with lists of `LatLng` to show them on a heatmap. If developers have a bunch of code that's expecting `LatLng` types and they now have a bunch of `WeightedLatLng` types instead that might be a pain to deal with.

Reply:
I don't follow; could you provide an example of code that would be easier with inheritance than with composition?

Reply:
I went ahead and changed it